commands/template_store_list.go

```diff
@@ -0,0 +1,175 @@
+	if clientErr != nil {
+		return nil, fmt.Errorf("error while requesting template list: %s\n", clientErr.Error())
+	}
+
+	if res != nil {
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("unexpected status code wanted: %d got: %d\n", http.StatusOK, res.StatusCode)
+		}
+
+		body, bodyErr := ioutil.ReadAll(res.Body)
+		if bodyErr != nil {
+			return nil, fmt.Errorf("error while reading data from templates body: %s\n", bodyErr.Error())
+		}
+
+		templatesInfo := []TemplateInfo{}
+		unmarshallErr := json.Unmarshal(body, &templatesInfo)
+		if unmarshallErr != nil {
+			return nil, fmt.Errorf("error while unmarshalling into templates struct: %s\n", unmarshallErr.Error())
+		}
+		return templatesInfo, nil
+	}
+	return nil, fmt.Errorf("error empty response body from: %s\n", DefaultTemplateList)
+}
+
+func formatCommandOutput(templates []TemplateInfo) string {
+	if source != allTemplates && source != incubatorRepo && source != officialRepo {
+		fmt.Printf("\n`The --source` flag needs to have value %s, %s or %s\n", allTemplates, incubatorRepo, officialRepo)
+		return ""
+	}
+	templaetsOutput := formatTemplatesOutput(templates, verbose, source)
+
+	return templaetsOutput
+}
+
+type TemplateInfo struct {
+	TemplateName string `json:"cli-lang"`
+	Platform     string `json:"platform"`
+	Language     string `json:"language"`
+	Source       string `json:"source"`
+	Description  string `json:"description"`
+	Repository   string `json:"repo"`
+	Official     string `json:"official"`
+}
+
+func formatTemplatesOutput(templates []TemplateInfo, verbose bool, source string) string {
+	var buff bytes.Buffer
+	lineWriter := tabwriter.NewWriter(&buff, 0, 0, 1, ' ', 0)
+
+	fmt.Fprintln(lineWriter)
+	if verbose {
+		formatVerboseOutput(lineWriter, templates, source)
+	} else {
+		formatBasicOutput(lineWriter, templates, source)
+	}
+	fmt.Fprintln(lineWriter)
+
+	lineWriter.Flush()
+
+	return buff.String()
+}
+
+func formatBasicOutput(lineWriter *tabwriter.Writer, templates []TemplateInfo, source string) {
+	fmt.Fprintf(lineWriter, "NAME\tSOURCE\tDESCRIPTION\n")
+	for _, template := range templates {
+		if source == incubatorRepo && template.Source == incubatorRepo {
+			fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+				template.TemplateName,
+				template.Source,
+				template.Description)
+		} else if source == officialRepo && template.Source == officialRepo {
+			fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+				template.TemplateName,
+				template.Source,
+				template.Description)
+		} else if source == allTemplates {
+			fmt.Fprintf(lineWriter, "%s\t%s\t%s\n",
+				template.TemplateName,
+				template.Source,
+				template.Description)
+		}
+	}
+}
+
+func formatVerboseOutput(lineWriter *tabwriter.Writer, templates []TemplateInfo, source string) {
+	fmt.Fprintf(lineWriter, "NAME\tLANGUAGE\tPLATFORM\tSOURCE\tDESCRIPTION\n")
+	for _, template := range templates {
+		if source == incubatorRepo && template.Source == incubatorRepo {
```

Comment:
Not sure why we need these three conditions.   What do they do and why?

Reply:
source removed ✔️